builtins.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env nix-shell
+#!nix-shell -i bash -p bash jq nix moreutils
+
+header="syn keyword nixNamespacedBuiltin contained"
+
+tmp_file="$(mktemp)"
+trap "rm $tmp_file" EXIT INT TERM QUIT
+
+nix __dump-language | jq -r '
+  def to_vim: keys | map("  \\ " + . + "\n") | .[];
+
+  [ (.builtins | to_vim)
+  , (.constants | del(.["true","false","null","builtins"]) | to_vim)
+  ] | add
+' > "$tmp_file"
+
+sed -E '/^'"$header"'/,/^$/ c __builtin_replacement__' syntax/nix.vim \
+  | sed '/^__builtin_replacement__/ r '"$tmp_file" \
+  | sed 's/^__builtin_replacement__/'"$header"'/' \
+  | sponge syntax/nix.vim
```

Comment:
Small nitpick because I hate tempfiles with a passion:

```suggestion
nix __dump-language | jq -r '
  def to_vim: keys | map("  \\ " + . + "\n") | .[];

  [ (.builtins | to_vim)
  , (.constants | del(.["true","false","null","builtins"]) | to_vim)
  ] | add
' | sed -n -i -e '/^'"$header"'/ {
  p; n; r /dev/stdin
  :l; n; /^$/!bl; b
}; p' syntax/nix.vim
```

With this, `moreutils` also shouldn't be needed.

Reply:
After reading the `sed` manpage I _think_ I understand this, but I also tested it manually and it appears to work fine.

Added you as co-author :)

Reply:
> After reading the `sed` manpage

Unfortunately, the manpage is a little bit incomplete, `info sed` is much more thorough and also lists *all* commands.